mapeditor/inspector/baseinspectoritemdelegate.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * baseinspectoritemdelegate.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#pragma once
+
+#include <QStyledItemDelegate>
+#include <QAbstractItemModel>
+
+class BaseInspectorItemDelegate : public QStyledItemDelegate
+{
+    Q_OBJECT
+public:
+    using QStyledItemDelegate::QStyledItemDelegate;
+
+    virtual void updateModelData(QAbstractItemModel * model, const QModelIndex & index) const {};
```

Comment:
if it's not intended to create objects of type BaseInspectorItemDelegate, then this method can be made pure virtual:
```suggestion
    virtual void updateModelData(QAbstractItemModel * model, const QModelIndex & index) const = 0;
```

Reply:
Objects of type BaseInspectorItemDelegate are not created, but not all derived classes implements this method. Would be it better to add empty implementation in those classes or leave this method as is?

Reply:
In such case empty implementation is fine. Unless you want to require subclasses to have explicit implementation.